File: gui/src/components/run/helpers.test.ts
import { describe, expect, it } from 'vitest'
import { compactArgs, pathFromArgs } from './helpers'

describe('run helpers', () => {
  it('shows only the path for edit args formatted as key=value pairs', () => {
    const args = 'newString="next,value",oldString="old,value",path=/tmp/file.go,replaceAll=false'

    expect(compactArgs(args)).toBe('/tmp/file.go')
    expect(pathFromArgs(args)).toBe('/tmp/file.go')
  })

  it('keeps bash command previews readable for key=value pairs', () => {
    expect(compactArgs('command=echo hello')).toBe('echo hello')
  })
})

File: tui/helpers_test.go
package tui

import "testing"

func TestFormatBriefArgsParsesJSONToolArgs(t *testing.T) {
	if got := formatBriefArgs("edit", `{"path":"/tmp/a.go","oldString":"a","newString":"b"}`); got != "/tmp/a.go" {
		t.Fatalf("edit args = %q, want path", got)
	}
	if got := formatBriefArgs("bash", `{"command":"go test ./..."}`); got != "go test ./..." {
		t.Fatalf("bash args = %q, want command", got)
	}
}

func TestFormatBriefArgsKeepsPairSyntax(t *testing.T) {
	if got := formatBriefArgs("edit", `path=/tmp/a.go,oldString=a,newString=b`); got != "/tmp/a.go" {
		t.Fatalf("edit args = %q, want path", got)
	}
}
